### PR TITLE
feat!: Update for new release

### PR DESCRIPTION
**Pull request by VALERA771**
## Description
**Describe the changes** 
No Christmas changes here. They'll be on other branch

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?** (if this is a feature change)

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

**Other information**:

<br />

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentations
<br />

## Submission checklist

- [x] I have checked the project can be compiled
- [ ] I have tested my changes and it worked as expected

### Patches (if there are any changes related to Harmony patches)
- [ ] I have checked no IL patching errors in the console

### Other
- [ ] Still requires more testing

### CODE REVIEW

File: EXILED/Exiled.API/Features/Items/MicroHid.cs [hunk 1]
```diff
@@ -49,23 +81,115 @@ public float Energy
...
+
         /// <summary>
         /// Starts firing the MicroHID.
         /// </summary>
-        public void Fire() => Base.Fire();
+        /// <param name="firingMode">Fire mode.</param>
+        public void Fire(MicroHidFiringMode firingMode = MicroHidFiringMode.PrimaryFire)
+        {
+            if (TryGetFireController(firingMode, out PrimaryFireModeModule module))
+                module.GetType().GetMethod("ServerFire", BindingFlags.Public | BindingFlags.DeclaredOnly)!.Invoke(module, Array.Empty<object>());
```
Comment (Misfiy) [inline EXILED/Exiled.API/Features/Items/MicroHid.cs:147]: Save it

**Review by Misfiy — CHANGES_REQUESTED** [review]

File: EXILED/Exiled.API/Features/Items/MicroHid.cs (hunk 1 above)
Comment (VALERA771) [reply EXILED/Exiled.API/Features/Items/MicroHid.cs:147]: MethodInfo?

Comment (louis1706) [reply EXILED/Exiled.API/Features/Items/MicroHid.cs:147]: yes

Comment (louis1706) [reply EXILED/Exiled.API/Features/Items/MicroHid.cs:147]: shouldn't this method call this instead ServerUpdateSelected(MicroHidPhase MicroHidPhase.Firing) from FiringModeControllerModule

File: EXILED/Exiled.Events/EventArgs/Player/UsingMicroHIDEnergyEventArgs.cs [hunk 2]
```diff
@@ -17,32 +17,24 @@ namespace Exiled.Events.EventArgs.Player
...
         /// <inheritdoc cref="IsAllowed" />
         /// </param>
-        public UsingMicroHIDEnergyEventArgs(Player player, MicroHIDItem microHIDitem, HidState currentState, float drain, bool isAllowed = true)
+        public UsingMicroHIDEnergyEventArgs(MicroHIDItem microHIDitem, float drain, bool isAllowed = true)
         {
-            Player = player;
             MicroHID = Item.Get<MicroHid>(microHIDitem);
-            CurrentState = currentState;
-            Drain = drain;
+            Drain = MicroHID.Energy - drain;
```
Comment (louis1706) [inline EXILED/Exiled.Events/EventArgs/Player/UsingMicroHIDEnergyEventArgs.cs:37]: are you sure of this it's fell like you made it wrong

Comment (VALERA771) [reply EXILED/Exiled.Events/EventArgs/Player/UsingMicroHIDEnergyEventArgs.cs:37]: Drain is a amount that will be removed. In method amount of new energy is saved. So yeah that should work

**Review by Misfiy — APPROVED** [review]

**Comment by ITeMbI4** [comment]
```[2024-12-22 19:43:45.497 +00:00] [ERROR] [Exiled.Events] Patching by attributes failed!
                                 HarmonyLib.HarmonyException: Patching exception in method virtual System.Boolean <_ProcessServerSideCode>d__50::MoveNext() ---> System.FormatException: Method virtual System.Boolean <_ProcessServerSideCode>d__50::MoveNext() cannot be patched. Reason: Invalid IL code in (wrapper dynamic-method) RoundSummary/<_ProcessServerSideCode>d__50:RoundSummary+<_ProcessServerSideCode>d__50.MoveNext_Patch0 (RoundSummary/<_ProcessServerSideCode>d__50): IL_03f0: brfalse   IL_0416
                                 
                                 
                                   at HarmonyLib.PatchFunctions.UpdateWrapper (System.Reflection.MethodBase original, HarmonyLib.PatchInfo patchInfo) [0x0008c] in <0020ee40255b4dc7a0eb8b2a76225b38>:0 
                                   at HarmonyLib.PatchClassProcessor.ProcessPatchJob (HarmonyLib.PatchJobs`1+Job[T] job) [0x000bb] in <0020ee40255b4dc7a0eb8b2a76225b38>:0 
                                    --- End of inner exception stack trace ---
                                   at HarmonyLib.PatchClassProcessor.ReportException (System.Exception exception, System.Reflection.MethodBase original) [0x0010f] in <0020ee40255b4dc7a0eb8b2a76225b38>:0 
                                   at HarmonyLib.PatchClassProcessor.Patch () [0x00093] in <0020ee40255b4dc7a0eb8b2a76225b38>:0 
                                   at Exiled.Events.Features.Patcher.PatchAll (System.Boolean includeEvents, System.Int32& failedPatch) [0x0005b] in <2e2e0b0568bf40d09295a06d0f792598>:0

**Pull request merged**